File: server/app/api/internal/types/order.go
package types

type CreateOrderReq struct {
	EventId        string   `json:"eventId"`
	TicketTierId   string   `json:"ticketTierId"`
	Quantity       int32    `json:"quantity"`
	TicketBuyerIds []string `json:"ticketBuyerIds"`
	AddressId      string   `json:"addressId,optional"`
	PayMethod      int32    `json:"payMethod,optional"`
	RequestId      string   `json:"requestId,optional"`
}

type GetOrderQueueStatusReq struct {
	QueueToken string `form:"queueToken" json:"queueToken,optional"`
}

type PayOrderReq struct {
	OrderNo   string `json:"orderNo"`
	PayMethod int32  `json:"payMethod,optional"`
	Channel   string `json:"channel,optional"`
}

type CancelOrderReq struct {
	OrderNo string `json:"orderNo"`
}

type ApplyRefundReq struct {
	OrderNo string `json:"orderNo"`
	Reason  string `json:"reason,optional"`
}

type ListOrderReq struct {
	Status   int32 `form:"status,optional" json:"status,optional"`
	Page     int32 `form:"page,optional" json:"page,optional"`
	PageSize int32 `form:"pageSize,optional" json:"pageSize,optional"`
}

type GetOrderDetailReq struct {
	OrderNo string `form:"orderNo" json:"orderNo,optional"`
}

type OrderBuyerInfo struct {
	Id     string `json:"id"`
	Name   string `json:"name"`
	IdCard string `json:"idCard"`
	Phone  string `json:"phone"`
}

type OrderTicketInfo struct {
	Id            string         `json:"id"`
	TicketBuyerId string         `json:"ticketBuyerId"`
	TicketCode    string         `json:"ticketCode"`
	QrCodeUrl     string         `json:"qrCodeUrl"`
	Status        int32          `json:"status"`
	SeatInfo      string         `json:"seatInfo"`
	VerifiedAt    string         `json:"verifiedAt"`
	Buyer         OrderBuyerInfo `json:"buyer"`
}

type OrderPaymentInfo struct {
	Id           string  `json:"id"`
	PaymentNo    string  `json:"paymentNo"`
	PayMethod    int32   `json:"payMethod"`
	Amount       float64 `json:"amount"`
	Status       int32   `json:"status"`
	TradeNo      string  `json:"tradeNo"`
	PaidAt       string  `json:"paidAt"`
	CreatedAt    string  `json:"createdAt"`
	CallbackData string  `json:"callbackData"`
}

type OrderRefundInfo struct {
	Id           string  `json:"id"`
	RefundNo     string  `json:"refundNo"`
	RefundAmount float64 `json:"refundAmount"`
	Status       int32   `json:"status"`
	Reason       string  `json:"reason"`
	RejectReason string  `json:"rejectReason"`
	TradeNo      string  `json:"tradeNo"`
	CreatedAt    string  `json:"createdAt"`
	RefundedAt   string  `json:"refundedAt"`
}

type OrderDeliveryInfo struct {
	TicketType     int32  `json:"ticketType"`
	DeliveryMethod string `json:"deliveryMethod"`
	AddressId      string `json:"addressId"`
	ReceiverName   string `json:"receiverName"`
	ReceiverPhone  string `json:"receiverPhone"`
	Province       string `json:"province"`
	City           string `json:"city"`
	District       string `json:"district"`
	Detail         string `json:"detail"`
	DeliveryStatus string `json:"deliveryStatus"`
}

type OrderSummary struct {
	Id             string  `json:"id"`
	OrderNo        string  `json:"orderNo"`
	EventId        string  `json:"eventId"`
	TicketTierId   string  `json:"ticketTierId"`
	EventTitle     string  `json:"eventTitle"`
	PosterUrl      string  `json:"posterUrl"`
	VenueName      string  `json:"venueName"`
	City           string  `json:"city"`
	EventStartTime string  `json:"eventStartTime"`
	EventEndTime   string  `json:"eventEndTime"`
	TicketTierName string  `json:"ticketTierName"`
	Quantity       int32   `json:"quantity"`
	UnitPrice      float64 `json:"unitPrice"`
	TotalAmount    float64 `json:"totalAmount"`
	Status         int32   `json:"status"`
	StatusText     string  `json:"statusText"`
	PayDeadline    string  `json:"payDeadline"`
	PaidAt         string  `json:"paidAt"`
	CancelledAt    string  `json:"cancelledAt"`
	CreatedAt      string  `json:"createdAt"`
	TicketType     int32   `json:"ticketType"`
}

type OrderDetail struct {
	Id             string            `json:"id"`
	OrderNo        string            `json:"orderNo"`
	UserId         string            `json:"userId"`
	EventId        string            `json:"eventId"`
	TicketTierId   string            `json:"ticketTierId"`
	EventTitle     string            `json:"eventTitle"`
	Description    string            `json:"description"`
	PosterUrl      string            `json:"posterUrl"`
	VenueName      string            `json:"venueName"`
	VenueAddress   string            `json:"venueAddress"`
	City           string            `json:"city"`
	EventStartTime string            `json:"eventStartTime"`
	EventEndTime   string            `json:"eventEndTime"`
	SaleStartTime  string            `json:"saleStartTime"`
	SaleEndTime    string            `json:"saleEndTime"`
	TicketTierName string            `json:"ticketTierName"`
	Quantity       int32             `json:"quantity"`
	UnitPrice      float64           `json:"unitPrice"`
	TotalAmount    float64           `json:"totalAmount"`
	Status         int32             `json:"status"`
	StatusText     string            `json:"statusText"`
	PayDeadline    string            `json:"payDeadline"`
	PaidAt         string            `json:"paidAt"`
	CancelledAt    string            `json:"cancelledAt"`
	CreatedAt      string            `json:"createdAt"`
	PurchaseLimit  int32             `json:"purchaseLimit"`
	NeedRealName   int32             `json:"needRealName"`
	TicketType     int32             `json:"ticketType"`
	Delivery       OrderDeliveryInfo `json:"delivery"`
	Tickets        []OrderTicketInfo `json:"tickets"`
	Payment        OrderPaymentInfo  `json:"payment"`
	Refund         OrderRefundInfo   `json:"refund"`
}

type CreateOrderResp struct {
	OrderNo     string `json:"orderNo"`
	QueueToken  string `json:"queueToken"`
	QueueStatus int32  `json:"queueStatus"`
	Message     string `json:"message"`
}

type OrderQueueStatusResp struct {
	QueueToken  string       `json:"queueToken"`
	OrderNo     string       `json:"orderNo"`
	QueueStatus int32        `json:"queueStatus"`
	Message     string       `json:"message"`
	Order       OrderSummary `json:"order,optional"`
}

type PayOrderResp struct {
	Success           bool             `json:"success"`
	PayForm           string           `json:"payForm"`
	Payment           OrderPaymentInfo `json:"payment"`
	OrderStatus       int32            `json:"orderStatus"`
	PaidAt            string           `json:"paidAt"`
	CheckoutUrl       string           `json:"checkoutUrl"`
	CheckoutSessionId string           `json:"checkoutSessionId"`
	SessionExpiresAt  string           `json:"sessionExpiresAt"`
}

type ApplyRefundResp struct {
	Success bool            `json:"success"`
	OrderNo string          `json:"orderNo"`
	Refund  OrderRefundInfo `json:"refund"`
}

type OrderListResp struct {
	Orders   []OrderSummary `json:"orders"`
	Total    int64          `json:"total"`
	Page     int32          `json:"page"`
	PageSize int32          `json:"pageSize"`
}

type OrderDetailResp struct {
	Order OrderDetail `json:"order"`
}

File: front/src/api/order.ts
import request from '@/utils/request'

export interface CreateOrderPayload {
  eventId: string
  ticketTierId: string
  quantity: number
  ticketBuyerIds: string[]
  addressId?: string
  payMethod?: number
  requestId?: string
}

export interface GetOrderQueueStatusParams {
  queueToken: string
}

export interface PayOrderPayload {
  orderNo: string
  payMethod?: number
  channel?: string
}

export interface CancelOrderPayload {
  orderNo: string
}

export interface ApplyRefundPayload {
  orderNo: string
  reason?: string
}

export interface ListOrderParams {
  status?: number
  page?: number
  pageSize?: number
}

export interface GetOrderDetailParams {
  orderNo: string
}

export interface OrderBuyerPayload {
  id: string
  name: string
  idCard: string
  phone: string
}

export interface OrderTicketPayload {
  id: string
  ticketBuyerId: string
  ticketCode: string
  qrCodeUrl: string
  status: number
  seatInfo: string
  verifiedAt: string
  buyer: OrderBuyerPayload
}

export interface OrderPaymentPayload {
  id: string
  paymentNo: string
  payMethod: number
  amount: number
  status: number
  tradeNo: string
  paidAt: string
  createdAt: string
  callbackData: string
}

export interface OrderRefundPayload {
  id: string
  refundNo: string
  refundAmount: number
  status: number
  reason: string
  rejectReason: string
  tradeNo: string
  createdAt: string
  refundedAt: string
}

export interface OrderDeliveryPayload {
  ticketType: number
  deliveryMethod: string
  addressId: string
  receiverName: string
  receiverPhone: string
  province: string
  city: string
  district: string
  detail: string
  deliveryStatus: string
}

export interface OrderSummaryPayload {
  id: string
  orderNo: string
  eventId: string
  ticketTierId: string
  eventTitle: string
  posterUrl: string
  venueName: string
  city: string
  eventStartTime: string
  eventEndTime: string
  ticketTierName: string
  quantity: number
  unitPrice: number
  totalAmount: number
  status: number
  statusText: string
  payDeadline: string
  paidAt: string
  cancelledAt: string
  createdAt: string
  ticketType: number
}

export interface OrderDetailPayload {
  id: string
  orderNo: string
  userId: string
  eventId: string
  ticketTierId: string
  eventTitle: string
  description: string
  posterUrl: string
  venueName: string
  venueAddress: string
  city: string
  eventStartTime: string
  eventEndTime: string
  saleStartTime: string
  saleEndTime: string
  ticketTierName: string
  quantity: number
  unitPrice: number
  totalAmount: number
  status: number
  statusText: string
  payDeadline: string
  paidAt: string
  cancelledAt: string
  createdAt: string
  purchaseLimit: number
  needRealName: number
  ticketType: number
  delivery: OrderDeliveryPayload
  tickets: OrderTicketPayload[]
  payment: OrderPaymentPayload
  refund: OrderRefundPayload
}

export interface CreateOrderResp {
  orderNo: string
  queueToken: string
  queueStatus: number
  message: string
}

export interface OrderQueueStatusResp {
  queueToken: string
  orderNo: string
  queueStatus: number
  message: string
  order?: OrderSummaryPayload
}

export interface PayOrderResp {
  success: boolean
  payForm: string
  payment: OrderPaymentPayload
  orderStatus: number
  paidAt: string
  checkoutUrl: string
  checkoutSessionId: string
  sessionExpiresAt: string
}

export interface ApplyRefundResp {
  success: boolean
  orderNo: string
  refund: OrderRefundPayload
}

export interface OrderListResp {
  orders: OrderSummaryPayload[]
  total: number
  page: number
  pageSize: number
}

export interface OrderDetailResp {
  order: OrderDetailPayload
}

export function createOrder(data: CreateOrderPayload) {
  return request.post<CreateOrderResp, CreateOrderResp>('/order/create', data)
}

export function getOrderQueueStatus(params: GetOrderQueueStatusParams) {
  return request.get<OrderQueueStatusResp, OrderQueueStatusResp>('/order/queue-status', {
    params,
  })
}

export function payOrder(data: PayOrderPayload) {
  return request.post<PayOrderResp, PayOrderResp>('/order/pay', data)
}

export function cancelOrder(data: CancelOrderPayload) {
  return request.post<{ success: boolean }, { success: boolean }>('/order/cancel', data)
}

export function applyRefund(data: ApplyRefundPayload) {
  return request.post<ApplyRefundResp, ApplyRefundResp>('/order/refund', data)
}

export function listOrders(params: ListOrderParams = {}) {
  return request.get<OrderListResp, OrderListResp>('/order/list', {
    params,
  })
}

export function getOrderDetail(params: GetOrderDetailParams) {
  return request.get<OrderDetailResp, OrderDetailResp>('/order/detail', {
    params,
  })
}
